fitlins/interfaces/nistats.py

```diff
@@ -21,11 +21,19 @@ def prepare_contrasts(contrasts, all_regressors):
 
     out_contrasts = []
     for contrast in contrasts:
-        # Fill in zeros
-        weights = np.array([
-            [row[col] if col in row else 0 for col in all_regressors]
-            for row in contrast['weights']
-            ])
+        # Are any necessary values missing for contrast estimation?
+        missing = any([[n for n, v in row.items()
+                        if v != 0 and n not in all_regressors]
+                       for row in contrast['weights']])
+        if missing:
+            weights = None
```

Comment:
If we continue, then the contrast list is just shorter, and you don't need to deal with `None`s.

```suggestion
            continue
```

Would this work? We'll need to make sure that there isn't some other place where we're generating a list that should match 1-1, but no longer is.

Reply:
Was this not a sufficient place to block the production contrasts involving missing regressors?